File: src/dp/ClimbingStairs_70.java
package dp;

public class ClimbingStairs_70 {

    public static void main(String[] args) {
        System.out.println(new ClimbingStairs_70().climbStairs(1));
        System.out.println(new ClimbingStairs_70().climbStairs(2));
        System.out.println(new ClimbingStairs_70().climbStairs(3));
        System.out.println(new ClimbingStairs_70().climbStairs(4));
    }

    public int climbStairs(int n) {
        if (n == 1)
            return 1;

        int prev = 0;
        int curr = 1;

        for (int i = 0; i < n; i++) {
            int next = prev + curr;
            prev = curr;
            curr = next;
        }
        return curr;
    }
}

File: src/dp/climbing_stairs_70.py
# dp
# dp[i] =  dp[i-2] + dp[i-1]


class Solution(object):
    def climbStairs(self, n):
        if n < 2:
            return 1

        dp = [1 for _ in range(n + 1)]

        for i in range(2, n + 1):
            dp[i] = dp[i - 2] + dp[i - 1]

        return dp[n]


print(Solution().climbStairs(1))
print(Solution().climbStairs(2))
print(Solution().climbStairs(3))
print(Solution().climbStairs(5))
